### PR TITLE
[C#] feat: add OpenAIModel unit test

### PR DESCRIPTION
**Pull request by kuojianlu**
## Linked issues

closes: #1115

## Details

Provide a list of your changes here. If you are fixing a bug, please provide steps to reproduce the bug.

#### Change details

> Describe your changes, with screenshots and code snippets as appropriate

**code snippets**:

**screenshots**:

## Attestation Checklist

- [ ] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (we use [TypeDoc](https://typedoc.org/) to document our code)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### Additional information

> Feel free to add other relevant information below

### CODE REVIEW

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/AITests/OpenAIModelTests.cs [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+        {
+            // Arrange
+            var options = new OpenAIModelOptions("test-key", "test-model");
+
+            // Act
+            try
+            {
+                new OpenAIModel(options);
+            }
+            catch (Exception ex)
```
Comment (swatDong) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/AITests/OpenAIModelTests.cs:29]: New warning on build. How about just call constructor and let it to throw as is.

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/AITests/OpenAIModelTests.cs [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+            var promptMock = new Mock<Prompt>(new List<PromptSection>(), -1, true, "\n\n");
+            promptMock.Setup((prompt) => prompt.RenderAsTextAsync(
+                It.IsAny<ITurnContext>(), It.IsAny<IMemory>(), It.IsAny<IPromptFunctions<List<string>>>(),
+                It.IsAny<ITokenizer>(), It.IsAny<int>(), It.IsAny<CancellationToken>())).ReturnsAsync(renderedPrompt);
+            var promptTemplate = new PromptTemplate("test-prompt", promptMock.Object);
+            var options = new AzureOpenAIModelOptions("test-key", "test-deployment", "https://test.openai.azure.com/")
+            {
+                CompletionType = CompletionConfiguration.CompletionType.Text
+            };
+            var openAIModel = new OpenAIModel(options);
```
Comment (swatDong) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/AITests/OpenAIModelTests.cs:95]: Using `TestLoggerFactory` and setting `LogRequests` to true can increase a few coverage (log part)

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/AITests/OpenAIModelTests.cs (hunk 1 above)
Comment (kuojianlu) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/AITests/OpenAIModelTests.cs]: Nice suggestion, updated.

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/AITests/OpenAIModelTests.cs (hunk 2 above)
Comment (kuojianlu) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI.Tests/AITests/OpenAIModelTests.cs]: Good idea, added.

**Comment by kuojianlu** [comment]
image

**Review by swatDong — APPROVED** [review]

**Pull request merged**